code/datums/riding.dm

```diff
@@ -34,16 +32,36 @@
 /datum/riding/proc/force_dismount(mob/living/M)
 	ridden.unbuckle_mob(M)
 
-//Override this to set your vehicle's various pixel offsets
-//if they differ between directions, otherwise use the
-//generic variables
 /datum/riding/proc/handle_vehicle_offsets()
+	var/passindex = 0
 	if(ridden.has_buckled_mobs())
 		for(var/m in ridden.buckled_mobs)
+			passindex++
 			var/mob/living/buckled_mob = m
-			buckled_mob.setDir(ridden.dir)
-			buckled_mob.pixel_x = generic_pixel_x
-			buckled_mob.pixel_y = generic_pixel_y
+			var/list/offsets = get_offsets(passindex)
+			dir_loop:
+				for(var/offsetdir in offsets)
+					if(offsetdir == ridden.dir)
+						var/list/diroffsets = offsets[offsetdir]
+						buckled_mob.pixel_x = diroffsets[1]
+						if(diroffsets.len == 2)
+							buckled_mob.pixel_y = diroffsets[2]
+						if(diroffsets.len == 3)
+							buckled_mob.layer = diroffsets[3]
+						break dir_loop
+
+
+//Override this to set your vehicle's various pixel offsets
+/datum/riding/proc/get_offsets(pass_index) // list(dir = x, y, layer)
+	switch(pass_index)
+		if(1)
+			return list("[NORTH]" = list(-10, -4), "[SOUTH]" = list(16, 3), "[EAST]" = list(-4, 30), "[WEST]" = list(4, -3))
```

Comment:
but most vehicles that have 4 passengers will be cars

reeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee ok